design-proposals/new-proposals-framework.md

```diff
@@ -0,0 +1,74 @@
+
+
+With this proposal, we aim to:
+
+- Optimize contributors' and reviewers' bandwidth and experience.
+- Strengthen SIG ownership and accountability.
+- Improve the alignment of enhancements with the project's goals and architecture.
+- Separate the enhancement process from community matters.
+
+# Goals
+
+1. Introduce a new repository, `kubevirt/enhancements`, to house all VEPs.
+2. Prioritize VEPs at the start of each development cycle to streamline review and focus efforts.
+3. Establish SIG-specific review processes with a dedicated reviewer for each VEP.
+4. Create issues to track the progress, maturity, and associated bugs for each accepted VEP.
+5. Establish ownership of the VEP and any additional SIG collaboration.
+
+## Non-Goals
+
+- Removing the central approval mechanism at this stage.
+- Full decentralization of VEP review and approval processes.
+
+## Repos
+
+- **Impacted Repository**: `kubevirt/enhancements` and kubevirt/community
+
+# Design
+
+The process includes the following key components:
+1. **VEP Creation**: VEP authors will initiate proposals via PRs to the `kubevirt/enhancements` repository.
+2. **SIG Review and Collaboration**: Each VEP will have a target SIG, and the SIG will assign a dedicated reviewer to oversee the proposal, collaborate with other SIGs as needed, and provide feedback or veto when necessary.
+3. **Centralized Prioritization**: At the start of each release cycle, a centralized mechanism will prioritize accepted VEPs, focusing community efforts on these.
+4. **Visibility and Tracking**: The Author of an accepted VEPs will open an issue to track their progress, maturity stages (alpha, beta, GA), and list the associated bugs. And user feedback
+5. *Single source of truth**: Each VEP will be the authoritative reference for the associated feature. This aligns with the Kubernetes KEP process. It will ensure that each enhancement
+Includes all the relevant information, including the design and the state.
+
+## Scalability
+
+This design allows scalability by:
+- Distributing review responsibility across SIGs while maintaining central oversight.
+- Providing a framework for gradual decentralization as SIGs mature.
+- Centralized tracking to set priorities and ensure alignment across the project.
+- Uniform process 
+
+# Approvers
+The following individuals are proposed as the initial approvers for the kubevirt/enhancements repository. These approvers will be responsible for ensuring VEPs meet the required standards, align with the project’s goals and best practices.
```

Comment:
How did we end up with this list?
What is the reasoning and the tradeoffs?

Reply:
This is only a suggested list.

As I mentioned in the document, the goal is to gradually transfer responsibility to the sigs as they mature and show commitment to the project as a whole, not only to the specific group.

To kickstart this process, we need a small group of people who will be committed to this process and who are not acting as SIG chairs.
I don't see a benefit in having a large list of approvers who are effectively inactive.

If anyone else would like to commit to this work, they are more than welcome to step up and add a name here.

Reply:
I'm also conflicted to whether we need a small list of approvers for this work instead of simply all of the sig approvers. Anyhow each VEP will be owned by a target SIG, as mentioned above.

I'd at least add here that we aim that in the future this work would be done by SIGs, which I see as part of their core responsibility.

Reply:
The goal is to eventually shift responsibilities toward the SIGs. However, this PR not only proposes changing the mechanics of how we manage proposals in KubeVirt but also asks the SIGs to adopt a different mindset.

As SIGs take on more control over VEPs, it is essential for the approvers to prioritize the benefit of the entire project over their downstream priorities or the interests of individual SIGs.
Unfortunately, there are many examples where this was not the case.

This is why I think we need a transition period that would allow SIGs to mature and adjust while keeping the centralized oversight.

FWIW, we could add here the current list of the community repo approvers, excluding those who are conflicted.